evadb/executor/llm_executor.py

```diff
@@ -0,0 +1,36 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Iterator
+
+from evadb.database import EvaDBDatabase
+from evadb.executor.abstract_executor import AbstractExecutor
+from evadb.models.storage.batch import Batch
+from evadb.plan_nodes.llm_plan import LLMPlan
+
+
+class LLMExecutor(AbstractExecutor):
+    def __init__(self, db: EvaDBDatabase, node: LLMPlan):
+        super().__init__(db, node)
+        self.llm_expr = node.llm_expr
+        self.alias = node.alias
+
+    def exec(self, *args, **kwargs) -> Iterator[Batch]:
+        child_executor = self.children[0]
+        for batch in child_executor.exec(**kwargs):
+            llm_result = self.llm_expr.evaluate(batch)
```

Comment:
Is the batch optimization done in the LLMExecutor and will be added in future PRs?

Reply:
Yes